### PR TITLE
Add profile option to latency benchmark

### PR DESCRIPTION
**Pull request by WoosukKwon**
This PR adds `--profile` option to benchmark latency. If enabled, we use PyTorch profiler to profile the GPU activities during the generation process.

### CODE REVIEW

File: benchmarks/benchmark_latency.py [hunk 1]
```diff
@@ -39,22 +37,29 @@ def main(args: argparse.Namespace):
...
-                     sampling_params=sampling_params,
-                     use_tqdm=False)
-
-        end_time = time.perf_counter()
-        latency = end_time - start_time
-        if profile:
-            torch.cuda.cudart().cudaProfilerStop()
-        return latency
+            with torch.profiler.profile(
+                    activities=[torch.profiler.ProfilerActivity.CUDA]) as p:
```
Comment (Yard1) [inline benchmarks/benchmark_latency.py:41]: I would suggest measuring CPU activity as well

Comment (WoosukKwon) [reply benchmarks/benchmark_latency.py]: Sounds good! Added. PTAL.

**Review by Yard1 — APPROVED** [review]
QQ, where will the profiler output be stored? Working dir?

**Comment by WoosukKwon** [comment]
@Yard1 Not really. It will be just printed. I think this is good for now?

**Comment by Yard1** [comment]
@WoosukKwon How about something like this:
```python
model_name = args.model.replace("/", "-")
profile_logdir_name = os.path.join(
    args.profile_logdir,
    f"{model_name}_tp-{args.tensor_parallel_size}_input-len{args.input_len}_output-len{args.output_len}_batch-size{args.batch_size}"
    .lstrip("-"))
with torch.profiler.profile(
    activities=[
        torch.profiler.ProfilerActivity.CPU,
        torch.profiler.ProfilerActivity.CUDA
    ],
    on_trace_ready=torch.profiler.tensorboard_trace_handler(
        profile_logdir_name),
    with_stack=True):
        ...
```

that way a JSON trace file will be saved which can be analyzed with tensorboard/perfetto (super useful!). `profile_logdir` will need to be added to CLI args.

**Comment by WoosukKwon** [comment]
@Yard1 Makes sense. But let's do it in another PR.

**Pull request merged**